app/main.py

```diff
@@ -96,3 +114,47 @@ async def get_subtype(
                 raise HTTPException(status_code=404, detail="Subtype not found")
 
     return {"message": result}
+
+@app.post("/signup", tags=["User"], status_code=201)
+async def signup(
+    cm: Annotated[ConnectionManager, Depends(get_connection_manager)],
+    credentials: HTTPBasicCredentials = Depends(auth)
+):
+    username = credentials.username
+
+    if not username:
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Missing email address!")
+
+    try:
+        with cm as connection_manager:
+            with connection_manager.get_cursor() as cursor:
+                logger.info(f"Creating user: {username}")
+                user = await new_user(cursor, username, FERTISCAN_STORAGE_URL)
+        return {"user_id": user.get_id()}
+
+    except Exception as e:
+        logger.error(f"Error occurred: {e}")
+        logger.error("Traceback: " + traceback.format_exc())
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Failed to create user!")
```

Comment:
just use cm directly instead of creating a connection_manager alias, or set the variable name in the function signature to connection_manager directly

Reply:
Using the cm like that makes sure that after exiting the context manager, resources are cleared automatically with our custom logic where we check if the app is in testing mode.

It is still possible to use it directly though. But we have to handle the clearing logic in the lifespan.

Reply:
I see, that looks ugly as hell though.

Reply:
I'll look for a better solution then.

Reply:
Don't worry about it. I understand the need. This is a good use case for the with statement.